mqtt/mqtt-bridge/src/upstream/rpc/mod.rs

```diff
@@ -110,17 +112,56 @@ impl Display for RpcCommand {
 /// response comes back as `mqtt3::Event` type which handled with the event
 /// handler.
 #[derive(Debug, Clone, Default)]
-pub struct RpcSubscriptions(Arc<Mutex<HashMap<String, CommandId>>>);
+pub struct RpcSubscriptions(Arc<Mutex<HashMap<String, VecDeque<CommandId>>>>);
```

Comment:
How many ids and what all operations do you expect here? Mainly ask if we can unify on Vec or VecDeque so we could pass a reference to the collection of CommandIds. Not a requirement, but might be nice :)

Reply:
I should not be the case at all! But for a number of reasons we cannot rely on EdgeHub to guaranty unique RPC request per topic.

I guess we cannot move out vec of CommandId out of shared collection while not holding a mutex. so making a copy of a small Vec may be a good enough.

Reply:
Sounds good to me, just want to avoid big copies 😸